### PR TITLE
Bugfix/jwhite242/versioninfo

### PR DESCRIPTION
**Pull request by jwhite242**
Update version info to sync up maestro's cli with pyproject.toml for single source of version.

### CODE REVIEW

File: maestrowf/__init__.py [hunk 1]
```diff
@@ -41,6 +41,8 @@
 import logging
 import sys
 
+from importlib_metadata import version  # Replace upon deprecation of python 3.7
```
Comment (FrankD412) [inline maestrowf/__init__.py:44]: Happen to know what this gets replaced with once 3.7 is deprecated? Is there an existing solution now?

Comment (FrankD412) [reply maestrowf/__init__.py:44]: Oh scratch that -- this is the backport. I see. I know it's a bit uglier but, mind replacing this with the usual
```python
try:
    from importlib.metadata import version
except ImportError as imp_err:
    from importlib_metadata import version
```

I think it makes it clearer that there's a backport and then when 3.7 is EOL'ed we can just remove the except.

File: pyproject.toml [hunk 2]
```diff
@@ -37,6 +37,7 @@ python = ">=3.7.1,<4.0"
 coloredlogs = "*"
 dill = "*"
 filelock = "*"
+importlib_metadata = "*"
```
Comment (FrankD412) [inline pyproject.toml:40]: This one is out of curiosity, but I think there's a way to conditionally install based on Python version if I'm not mistaken?

**Review by FrankD412 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (jwhite242) [reply pyproject.toml:40]: There is, but it's already there from some other dependency we have; happy to make it conditional if the dependencies can still override this.

File: maestrowf/__init__.py (hunk 1 above)
Comment (jwhite242) [reply maestrowf/__init__.py]: Done.

**Review by FrankD412 — APPROVED** [review]

**Pull request merged**